google-cloud-trace/lib/google/cloud/trace/faraday_middleware.rb

```diff
@@ -40,9 +40,14 @@ class FaradayMiddleware < Faraday::Middleware
         # end
         # ```
 
-        def initialize app, enable_cross_project_tracing: false
-          super(app)
+        def initialize app, opts = {}, enable_cross_project_tracing: false
+          # Allow enable_cross_project_tracing to be passed in either in an
+          # options hash or as a keyword argument. This may seem redundant, but
+          # it's necessary to work around a Faraday bug where the `new` class
```

Comment:
Can you link to the bug?

Reply:
I looked at this more closely and realized that it's actually an intentional choice in Faraday, to force all arguments to be positional. So we can just follow suit and use only an opts hash instead of checking both. This will work on Ruby 3 because Faraday converts keyword arguments to an opts hash using `ruby2_keywords`.  (And it works on Ruby 2 because Ruby automatically performs that conversion.)